lib/ui/fixtures/ui_test.dart

```diff
@@ -73,3 +74,60 @@ Future<void> encodeImageProducesExternalUint8List() async {
 void _encodeImage(Image i, int format, void Function(Uint8List result))
   native 'EncodeImage';
 void _validateExternal(Uint8List result) native 'ValidateExternal';
+
+@pragma('vm:entry-point')
+Future<void> pumpImage() async {
+  final Completer<Image> completer = Completer<Image>();
+  const int width = 8000;
+  const int height = 8000;
+  decodeImageFromPixels(
+    Uint8List.fromList(List<int>.filled(width * height * 4, 0xFF)),
+    width,
+    height,
+    PixelFormat.rgba8888,
+    (Image image) => completer.complete(image),
+  );
+
+  final Image image = await completer.future;
+
+  final FrameCallback renderBlank = (Duration duration) {
+    final PictureRecorder recorder = PictureRecorder();
+    final Canvas canvas = Canvas(recorder);
+    canvas.drawRect(Rect.largest, Paint());
+    final Picture picture = recorder.endRecording();
+
+    final SceneBuilder builder = SceneBuilder();
+    builder.addPicture(Offset.zero, picture);
+
+    final Scene scene = builder.build();
+    window.render(scene);
+    scene.dispose();
+    window.onBeginFrame = (Duration duration) => _done();
```

Comment:
Should `_done` go on `window.onDrawFrame`. The docs say that that will flush the microtask queue before running it, which may reduce flakiness?

Reply:
Looks like that worked - just had to put it in onDrawFrame on the next frame.